yt_dlp/extractor/common.py

```diff
@@ -2613,18 +2613,31 @@ def extract_common(source):
                 if segment_duration:
                     ms_info['segment_duration'] = float(segment_duration)
 
+            def parse_range(byte_range):
+                if isinstance(byte_range, str):
+                    splitted_byte_range = byte_range.split('-')
+                    if len(splitted_byte_range) == 2:
+                        return {
+                            'start': int(splitted_byte_range[0]),
+                            'end': int(splitted_byte_range[1]) + 1,
+                        }
```

Comment:
The PR parses the byte range in order to fit what is expected by the fragment downloader (where the 1 added to the range end is subtracted). However, the DASH spec (ISO 23009) specifically says that the range values are strings that conform to the HTTP specification for use in the `Range` header, so the processing here is not strictly necessary. Instead, the range could be passed into the HTTP header in the DASH downloader, as in https://github.com/ytdl-org/youtube-dl/pull/30279/.

On a style note, `splitted` isn't a normal English formation. The past participle of `split` is also `split`, and so is the past tense: "the range was split in two", "a colon never split the start and end of the range",

Reply:
Hello @dirkf! Thank you for your feedback!

Although it's possible to pass a range string directly to http headers, I decided to parse it not only because the fragment downloader doesn't support range strings. I found this in some MPD files:
```xml
<SegmentList timescale='1000' duration='4000'>
  <Initialization sourceURL='https://example.com/2160p.mp4' range='36-800'/>
  <SegmentURL media='https://example.com/2160p.mp4' mediaRange='801-379516'/>
  <SegmentURL media='https://example.com/2160p.mp4' mediaRange='379517-656206'/>
  <!-- SegmentURLs -->
</SegmentList>
```
So the same file is referenced several times with different ranges. In this case we can optimize the extractor by merging consecutive fragments where the URLs are the same and the start of the range is equal to the end of the previous range. In my case one file was referenced about 100 times and I believe by replacing 100 small requests with 1 large we could significantly reduce the downloading time. I hope to submit a PR with this optimization sometime, that's why I would like to stay with current `byte_range` format. What do you think?

Thank you for your note on language! I'm really appreciate it because I'm actively learning English now) I copied that variable name from `downloader/hls.py` and it seems there are several more variables with `splitted_` in their names. Maybe we can merge it as is and then correct all such spelling mistakes?) I can do it a bit later.

Reply:
Hello @dirkf! I added the optimization described above and fixed the spelling mistake. Could you please review the updated PR?